### PR TITLE
CB-18626 Enable dedicated storage account for azure E2E with SINGLE_WITH_DEDICATED_STORAGE_ACCOUNT resource group usage

### PR DESCRIPTION
**Pull request by aszegedi**
We have implemented a special single resource group usage flag for AZURE: `SINGLE_WITH_DEDICATED_STORAGE_ACCOUNT`. This kind of single resource group usage will reduce the execution time of the tests and might add to stability.

### CODE REVIEW

**Comment by aszegedi** [comment]
Prerequisite PR is available at Ansible Playbooks repository as [CB-18626 Enable dedicated storage account for azure E2E with SINGLE_WITH_DEDICATED_STORAGE_ACCOUNT resource group usage #198](https://github.com/hortonworks/cloudbreak-ansible-playbooks/pull/198)

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java [hunk 1]
```diff
@@ -222,7 +223,7 @@ public void testCreateDistroXWithEncryptedVolumesInSingleRG(TestContext testCont
 
         testContext
                 .given(EnvironmentNetworkTestDto.class)
-                    .withServiceEndpoints()
+                    .withServiceEndpoints(ServiceEndpointCreation.DISABLED)
```
Comment (lnardai) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java:226]: Why was this required?

Comment (aszegedi) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java:226]: Because of at [AZURE provider](https://github.com/hortonworks/cloudbreak/blob/master/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/cloud/v4/azure/AzureCloudProvider.java#L226-L234):
```
@Override
public ServiceEndpointCreation serviceEndpoint() {
    ServiceEndpointCreation serviceEndpointCreation = ResourceGroupTest.isSingleResourceGroup(getTestParameter().get(ResourceGroupTest.AZURE_RESOURCE_GROUP_USAGE))
                        ? ServiceEndpointCreation.ENABLED_PRIVATE_ENDPOINT
                        : ServiceEndpointCreation.DISABLED;
    LOGGER.debug("Azure service endpoint creation: {}", serviceEndpointCreation);
    return serviceEndpointCreation;
}
```
for `SINGLE` resource group usage the `ServiceEndpointCreation` is going to be `ENABLED_PRIVATE_ENDPOINT`. That is causing issue in case of a general AZURE E2E test: `TestFailException: Cluster 'azure-test-c8fdbb933939455a9' entered into creation failed state. Status: '{status=CREATE_FAILED}' statusReason: '{statusReason=The provided private DNS zone id  for service Microsoft.DBforPostgreSQL/servers is not a valid azure resource id.}'`

**Review by sodre90 — APPROVED** [review]
